RedHat-UX/red-hat-design-system
# fix(secondary-nav): update rh-context-provider styles

## What I did

Changes to `<rh-secondary-nav>`:
 - Updated styles for `rh-context-provider` in shadowroot to ensure cta centering.
 - **BREAKING**: Fixed incorrect color map for dark variant `color-palette="darker"` to `color-palette="dark"`
 
Changes to `colorContextController`:
 - Added `light` and `dark` context-color css rules to match surface tokens `--rh-color-surface-light` and `--rh-color-surface-dark`

Closes https://github.com/RedHat-UX/red-hat-design-system/issues/657


## Testing Instructions

1. View deploy preview
2. Ensure navbar slotted CTA is vertically centered 
3. Ensure cta on lighter variant nav bark is on `lighter` surface when in navbar and `lightest` when in mobile menu dropdown. 
4. Ensure cta on dark variant nav bark is on `dark` surface when in navbar and `lightest` when in mobile menu dropdown.
